### ISSUE TITLE
process ends unexpectedly with large padded conv1d

### ISSUE THREAD

**Issue by danFromTelAviv** [pytorch/pytorch#18652]
## 🐛 Bug
I am trying to train a production size network that has a set of large convolution filters ( millions of parameters , kernel size of 20+ )
When I train with a smaller dev version of the network - still tens of millions of parameters everything is fine. 
When I inspect the gpu performance I can see that the ram is fine. The issue seems to be a peak in the copy. More interestingly it looks likes the copy gets stuck on zero for a moment and then peaks too high which causes the process to die. 
![image](https://user-images.githubusercontent.com/31417918/55275051-459ca300-52f1-11e9-843c-a10cbf69c7b9.png)

## To Reproduce

1.create a net with very large convs (10 convs of 500 features and kernel which is 20 timesteps long  I think should be enough) with padding of 10 on each side. 
2. I am training on a ctc loss but I don't think thats the issue since the copy is whats "stalling".  The sequence len can reach 1000 regularly. 

 - PyTorch Version (e.g., 1.0): 1.0.1
 - OS (e.g., Linux): windows 10 
 - How you installed PyTorch (`conda`, `pip`, source): pip
 - Build command you used (if compiling from source): 
 - Python version: 3.6.3
 - CUDA/cuDNN version: 9.0
 - GPU models and configuration: gtx 1070 ti 8gb
 - Any other relevant information:

**Comment by danFromTelAviv** [comment]
after some more debugging to try to identify the issue - it looks like it happens when I decrease batch size from 8 to 4??

**Comment by peterjc123** [comment]
Don't use taskmgr for debugging. Use nvidia-smi instead. And could you please set the num_workers of the DataLoader to zero? Please provide the minimal reproducible code so that I can look into it.

**Comment by danFromTelAviv** [comment]
I'm not using the standard data loader. I have my own generator. I load all of the data for a mini epoch in parallel in advance of the next mini epoch so all of the data should be in the ram when these jumps in copy occur. 

I am not really sure how to use nvidia smi to profile the gpu. Iv'e never had to. It looks like it just prints out the current status of the gpu and closes ( it doesn't stay and update - like htop does ). 
Do you want me to reproduce and try to get a snapshot while its stalling and right after it peaks?

As for the reproducible code - this is a bit hard since it requires sharing the entire training procedure and net... and possibly some samples. Unfortunately, I can't do that... 
I can try to create a similar net that has the same issue - if I succeed ill post it.

**Comment by peterjc123** [comment]
You can use `nvida-smi -dmon` or `nvida-smi -l`.

**Comment by danFromTelAviv** [comment]
# gpu   pwr  temp    sm   mem   enc   dec  mclk  pclk
# Idx     W     C     %     %     %     %   MHz   MHz
    0    42    55     5     0     0     0  3802  1607
    0    42    55     5     0     0     0  3802  1607
    0    42    55     4     0     0     0  3802  1607
    0   168    59    67    21     0     0  3802  1771
    0   167    61    78    17     0     0  3802  1771
    0   151    61    87    26     0     0  3802  1771
    0   141    62    85    24     0     0  3802  1733
    0   135    63    84    21     0     0  3802  1784
    0    43    59     0     0     0     0  3802  1607
    0    44    58     0     0     0     0  3802  1607
    0    43    58     0     0     0     0  3802  1607
    0    43    58     0     0     0     0  3802  1607
    0    43    58     0     0     0     0  3802  1607
    0    43    58     0     0     0     0  3802  1607
    0    43    58     0     0     0     0  3802  1607
    0    43    57     0     0     0     0  3802  1607
    0    43    58     0     0     0     0  3802  1607
    0    43    58     0     0     0     0  3802  1607
    0    43    58     0     0     0     0  3802  1607
    0    43    58     0     0     0     0  3802  1607
    0    43    58     0     0     0     0  3802  1607
    0    43    58     0     0     0     0  3802  1607
    0    43    58     0     0     0     0  3802  1607
    0    43    58     0     0     0     0  3802  1607
    0    43    58     0     0     0     0  3802  1607
    0    43    58     0     0     0     0  3802  1607
    0    43    58     0     0     0     0  3802  1607
    0    43    58     0     0     0     0  3802  1607
    0    43    58     0     0     0     0  3802  1607
    0    43    58     0     0     0     0  3802  1607
    0    43    59     0     0     0     0  3802  1607
    0    43    58     0     0     0     0  3802  1607
    0    43    59     0     0     0     0  3802  1607
    0    43    59     0     0     0     0  3802  1607
    0    43    59     0     0     0     0  3802  1607
    0    43    59     0     0     0     0  3802  1607
    0    43    59     0     0     0     0  3802  1607
    0    43    59     0     0     0     0  3802  1607
    0    43    59     0     0     0     0  3802  1607
    0    43    59     0     0     0     0  3802  1607

**Comment by danFromTelAviv** [comment]
```
class Conv1dSamePad(nn.Module):
    def __init__(self, in_channels, out_channels, filter_len, **kwargs):
        super(Conv1dSamePad, self).__init__()
        self.filter_len = filter_len
        self.conv = nn.Conv1d(in_channels, out_channels, filter_len, padding=(self.filter_len // 2), **kwargs)
        nn.init.xavier_uniform_(self.conv.weight)

    def forward(self, x):
        if self.filter_len % 2 == 1:
            return self.conv(x)
        else:
            return self.conv(x)[:, :, :-1]

class ReproduceNet(nn.Module):
    def __init__(self, num_classes):
        super(ReproduceNet, self).__init__()
        self.num_classes = num_classes
        self.first_conv = Conv1dSamePad(13, 1000, 13)
        self.convs = nn.ModuleList(
            [nn.Sequential(Conv1dSamePad(1000, 1000, idx + 10), nn.BatchNorm1d(1000),
                           nn.ReLU(inplace=True)) for idx in range(15)])
        self.last_conv = nn.Sequential(Conv1dSamePad(1000, num_classes, 1))
        self.logsoftmax = nn.LogSoftmax(dim=1)

    def forward(self, x):
        x = self.first_conv(x)
        for idx, conv in enumerate(self.convs):
            x = conv(x) 
        return self.logsoftmax(self.last_conv(x))

```

**Comment by danFromTelAviv** [comment]
@peterjc123 have you been able to reproduce the error locally with the code I provided?

**Comment by iremyux** [comment]
Hi @danFromTelAviv , 

I know it has been years now, but I am curious to know if you are still encountering the same issue. Did this problem disappear afterwards or is it still present? I am asking to see if this issue can be closed.

**Comment by iremyux** [comment]
I am closing this issue due to lack of activity. Please feel free to open a new issue in the future if you encounter any problems or have feature requests.
